library/random/formatter/shared/interface.rb

```diff
@@ -0,0 +1,33 @@
+require_relative '../../../../spec_helper'
+
+describe :random_formatter_interface, shared: true do
+  it "calls the method #bytes for 4 blocks of 4 bytes" do
+    bytes = mock('bytes')
+    bytes.extend(Random::Formatter)
+    bytes.should_receive(:bytes).with(4).exactly(4).and_return("\x00".b * 4)
+    bytes.send(@method)
+  end
```

Comment:
It seems to me like an implementation detail - in what way the `#bytes` method is called.

Moreover these 4x4 calls approach is specific to the `#alphanumeric` method e.g. `#base64` calls by default `#bytes` only once:

```ruby
obj = Object.new
def obj.bytes(n)
   puts "#bytes(#{n})"
   "\x00".b*n
end
obj.singleton_class.include(Random::Formatter)

obj.base64
#bytes(16)
# => "AAAAAAAAAAAAAAAAAAAAAA=="
```

And `#alphanumeric` behaviour depends on `n`:

```ruby
obj.alphanumeric(8)
#bytes(4)
#bytes(4)
=> "AAAAAAAA"
```

So I would check only that the interface between the `Random::Formatter` module and a class that includes it is just a single method `#bites`, and nothing more is required.